### PR TITLE
Cmac/BFD-650 unit tests v2 patient resource

### PR DESCRIPTION
**Pull request by mackec1**
### Change Details

Modified all changes to leverage FHIR v4 deep comparisons of Patient resource.

### Acceptance Validation

Tests run successfully.
Tests use FHIR deep comparison to check values and element identifiers (url).

### Feedback Requested

Tests make sense and adequately verify functionality of FHIR v4 Patient resource.

### External References

- [BFD-650](https://jira.cms.gov/browse/BFD-650)

### Security Implications

N/A

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java [hunk 1]
```diff
@@ -66,225 +77,296 @@ public void setup() {
...
+
+  private void createPatient(RequestHeaders reqHeaders) {
+    patient = BeneficiaryTransformerV2.transform(new MetricRegistry(), beneficiary, reqHeaders);
   }
 
   @After
   public void tearDown() {
     beneficiary = null;
+    patient = null;
   }
```
Comment (rtravitz) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java:92]: Is this `After` function needed? Like what's the consequence for not setting these variables back to null?

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java [hunk 2]
```diff
@@ -66,225 +77,296 @@ public void setup() {
...
 
   /**
-   * @return the {@link StaticRifResourceGroup#SAMPLE_A} {@link Beneficiary} record, with the {@link
-   *     Beneficiary#getBeneficiaryHistories()} and {@link
-   *     Beneficiary#getMedicareBeneficiaryIdHistories()} fields populated.
+   * Verifies that {@link gov.cms.bfd.server.war.r4.providers.BeneficiaryTransformerV2} works
+   * correctly when passed a {@link Beneficiary} with includeAddressFields header values.
    */
+  @Test
+  public void shouldMatchAddressWithAddrHeader() {
```
Comment (rtravitz) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java:320]: More just for my knowledge: what's going on with needing a matching with address header in addition to matching without an address header? Looks like the serialized sample A output only has state and postal code, so I was interested to see this extra test.

**Review by rtravitz — APPROVED** [review]
Coverage of the serialized output is complete, how the tests are constructed makes sense to me, and they all pass. Nice ✨

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java (hunk 2 above)
Comment (mackec1) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java:320]: So the default will be to only return state and postal; however, if the RequestHeader has include address set to true, we return the entire address data.  So there is one test for the default, and one test for the entire address info.

**Review by Dranith — APPROVED** [review]
Looks good

**Review by cbrunefearless — APPROVED** [review]
LGTM

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java (hunk 1 above)
Comment (mackec1) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java:92]: I thought I had replied to this, but I guess not; just being anal and making sure state/data is cleared after every test.

**Pull request merged**